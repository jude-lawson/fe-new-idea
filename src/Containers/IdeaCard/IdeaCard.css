.idea--card-container {
  margin: 3rem 6rem;
  border: 3px solid var(--primary-red);
  padding: 2rem;
  background: #fff;
  position: relative;
  border-radius: 1rem;
  min-height: 12rem;
}

.card--rmv-btn {
  position: absolute;
  top: 2rem;
  right: 2rem;
  border: 2px solid var(--primary-red);
  padding: 4px 8px;
  border-radius: 10px;
  color: var(--primary-blue);
}

.card--rmv-btn:hover {
  background: var(--primary-red);
  color: var(--primary-white);
  border: 2px solid var(--primary-blue);
}

.card--comment-btn {
  position: absolute;
  bottom: 2rem;
  right: 2rem;
  border: 2px solid var(--primary-red);
  padding: 4px 8px;
  border-radius: 10px;
  color: var(--primary-blue);
}

.card--comment-btn:hover {
  background: var(--primary-blue);
  color: var(--primary-white);
  border: 2px solid var(--primary-red);
}