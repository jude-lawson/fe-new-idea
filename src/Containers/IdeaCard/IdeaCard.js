import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import './IdeaCard.css';

export const IdeaCard = (props) => {
  const {title, body} = props;
  return (
    <article className="idea--card-container">
      <button className="card--rmv-btn">REMOVE</button>
      <h1 className="idea--card-title">{title}</h1>
      <p className="idea--card-body">{body}</p>
      <button className="card--comment-btn">COMMENT</button>
    </article>
  );
};

IdeaCard.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string
};
