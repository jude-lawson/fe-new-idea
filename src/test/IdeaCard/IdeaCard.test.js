import React from 'react';
import { shallow } from 'enzyme';
import { IdeaCard, mapStateToProps } from '../../Containers/IdeaCard/IdeaCard';

describe('IdeaCard', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<IdeaCard />);

    expect(wrapper).toMatchSnapshot();
  });
});

