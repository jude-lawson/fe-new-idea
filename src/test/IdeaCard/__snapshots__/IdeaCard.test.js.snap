// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="card--rmv-btn"
>
          REMOVE
</button>,
        <h1
          className="idea--card-title"
/>,
        <p
          className="idea--card-body"
/>,
        <button
          className="card--comment-btn"
>
          COMMENT
</button>,
      ],
      "className": "idea--card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "REMOVE",
          "className": "card--rmv-btn",
        },
        "ref": null,
        "rendered": "REMOVE",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "idea--card-title",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "idea--card-body",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COMMENT",
          "className": "card--comment-btn",
        },
        "ref": null,
        "rendered": "COMMENT",
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="card--rmv-btn"
>
            REMOVE
</button>,
          <h1
            className="idea--card-title"
/>,
          <p
            className="idea--card-body"
/>,
          <button
            className="card--comment-btn"
>
            COMMENT
</button>,
        ],
        "className": "idea--card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "REMOVE",
            "className": "card--rmv-btn",
          },
          "ref": null,
          "rendered": "REMOVE",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "idea--card-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "idea--card-body",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COMMENT",
            "className": "card--comment-btn",
          },
          "ref": null,
          "rendered": "COMMENT",
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
